kotlin/find-connected-features-in-utility-networks/src/main/java/com/esri/arcgisruntime/sample/findconnectedfeaturesinutilitynetworks/MainActivity.kt

```diff
@@ -0,0 +1,458 @@
+              addUtilityElementToMap(identifiedFeature, mapPoint, utilityElement)
+            }
+            else -> error("Unexpected utility network source type!")
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Add utility element to either the starting locations or barriers list and add a graphic
+   * representing it to the graphics overlay.
+   */
+  private fun addUtilityElementToMap(
+    identifiedFeature: ArcGISFeature,
+    mapPoint: Point?,
+    utilityElement: UtilityElement
+  ) {
+    graphicsOverlay.graphics.add(
+      Graphic(
+        GeometryEngine.nearestCoordinate(
+          identifiedFeature.geometry,
+          mapPoint
+        ).coordinate
+      ).apply {
+        // add the element to the appropriate list (starting locations or barriers), and add the
+        // appropriate symbol to the graphic
+        symbol = if (startingLocationsRadioButton.isChecked) {
+          startingLocations.add(utilityElement)
+          startingPointSymbol
+        } else {
+          barriers.add(utilityElement)
+          barrierPointSymbol
+        }
+      })
+  }
+
+  /**
+   * Uses a UtilityNetworkSource to create a UtilityElement object out of an ArcGISFeature.
+   *
+   * @param identifiedFeature an ArcGISFeature object that will be used to create a UtilityElement
+   * @param networkSource     the UtilityNetworkSource to which the created UtilityElement is associated
+   * @return the created UtilityElement
+   */
+  private fun createUtilityElement(
+    identifiedFeature: ArcGISFeature,
+    networkSource: UtilityNetworkSource
+  ) {
+    // find the code matching the asset group name in the feature's attributes
+    val assetGroupCode =
+      identifiedFeature.attributes[identifiedFeature.featureTable.subtypeField] as Int
+
+    // find the network source's asset group with the matching code
+    networkSource.assetGroups.filter { it.code == assetGroupCode }[0].assetTypes
+      // find the asset group type code matching the feature's asset type code
+      .filter { it.code == identifiedFeature.attributes["assettype"].toString().toInt() }[0]
+      .let { utilityAssetType ->
+        // get the list of terminals for the feature
+        val terminals = utilityAssetType.terminalConfiguration.terminals
+        // if there is only one terminal, use it to create a utility element
+        when (terminals.size) {
+          1 -> {
+            utilityNetwork.createElement(identifiedFeature, terminals[0])
+          }
+          // if there is more than one terminal, prompt the user to select one
+          else -> {
+            // get a list of terminal names from the terminals
+            val terminalNames = utilityAssetType.terminalConfiguration.terminals.map { it.name }
+
+            AlertDialog.Builder(this).apply {
+              setTitle("Select utility terminal:")
+              setItems(terminalNames.toTypedArray()) { _, which ->
```

Comment:
Might be worth to add a comment here, something like:
When a terminal is selected, create a utility element from it, add it to the map and show its name in the UI.